website/content/docs/concepts/plugins/storage/host-volumes.mdx

```diff
@@ -0,0 +1,387 @@
+    stderr "deleting volume at $path"
+    delete_volume "$path" ;;
+  *)
+    echo "unknown operation: $1"
+    exit 1 ;;
+esac
+```
+
+</Tab>
+</Tabs>
+
+## Specification
+
+A plugin is registered with Nomad if it:
+
+* Is an executable file (a script or binary)
+* Is located in the [`client.host_volume_plugin_dir`][plugin_dir] directory
+* Responds appropriately to a `fingerprint` call
+
+### Operations
+
+To fully manage the lifecycle of a volume, plugins must fulfill all of the
+following operations:
+
+* [fingerprint](#fingerprint)
+* [create](#create)
+* [delete](#delete)
+
+Nomad passes the operation as the first positional argument to the plugin.
+That and other information are passed as environment variables. Environment
+variables are prefixed with `"DHV_"` (i.e. Dynamic Host Volume).
+
+Some variables may be required for certain plugins (namely, `DHV_CAPACITY_*`
+for plugins that can restrict size). Most are for plugin author convenience.
+
+#### fingerprint
+<blockquote style={{borderLeft: "solid 1px #00ca8e"}}>
+
+Nomad calls `fingerprint` when the client agent starts (or is reloaded with a
+SIGHUP) to discover valid plugins. The returned "version" is used to register
+the plugin on the Nomad node for volume scheduling.
+
+**CLI arguments:** `$1=fingerprint`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=fingerprint
+```
+
+**Expected stdout:**
+
+```
+{"version": "0.0.1"}
+```
+
+**Requirements:**
+
+* Must complete within 5 seconds, or Nomad will kill it
+  * should be much faster, as no actual work should be done
+* "version" value must be valid per the [hashicorp/go-version][go-version]
+  golang package
+
+</blockquote>
+
+#### create
+<blockquote style={{borderLeft: "solid 1px #00ca8e"}}>
+
+Nomad calls `create` when you run [`nomad volume create`][cli-create] CLI or
+use the [create API][api-create]).
+
+You can run `create` again for the same volume if you include the volume
+`id` in the volume specification.
+
+Nomad also calls `create` when the agent starts, for each volume that was
+previously created on the node, so plugins can ensure the volumes are available
+after an agent restart or host reboot.
+
+**CLI Arguments:** `$1=create`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=create
+DHV_HOST_PATH=/suggested/path/to/place/volume
+DHV_NODE_ID={Nomad node ID}
+DHV_NODE_POOL={Nomad node pool}
+DHV_NAMESPACE={volume namespace}
+DHV_VOLUME_NAME={name from the volume specification}
+DHV_VOLUME_ID={volume ID assigned by Nomad}
+DHV_CAPACITY_MIN_BYTES={capacity_min from the volume spec}
+DHV_CAPACITY_MAX_BYTES={capacity_max from the volume spec}
+DHV_PARAMETERS={json of parameters from the volume spec, or: null}
+DHV_PLUGIN_DIR={path to directory containing plugins}
+```
+
+**Expected stdout:**
+
+```
+{"path": "/path/to/created/volume", "bytes": 50000000}
+```
+
+**Requirements:**
+
+* Must complete within 60 seconds, or Nomad will kill it
+* Must be idempotent - repeat runs produce the same result
```

Comment:
Maybe explicitly say "for the same volume ID" and/or "for the same arguments"? It's not just that we're idempotent but that we'll update the volume too.